reference.rb

```diff
@@ -0,0 +1,66 @@
+class MoviesController < ApplicationController
+  def index
+    @movies = Movie.order(upvotes: :desc)
+  end
+
+  def show
+    id = params[:id]
+    @movie = Movie.find(id)
+  end
+
+  def new
+    @movie = Movie.new
+  end
+
+  def create
+    @movie = Movie.create(movie_params[:movie])
+    if @movie.save
+      redirect_to movie_path(@movie)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    id = params[:id]
+    @movie = Movie.find(id)
+  end
+
+  def update
+    id = params[:id]
+    @movie = Movie.find(id)
+    @movie.update(
+    name: movie_params[:movie][:name],
+    director: movie_params[:movie][:director],
+    description: movie_params[:movie][:description]
+    )
+    redirect_to movie_path(@movie)
+  end
+
+  def destroy
+    id = params[:id]
+    Movie.destroy(id)
+    redirect_to movies_path
+  end
+
+  def upvote
+    id = params[:id]
+    @movie = Movie.find(id)
+    current_score = Movie.find(id).upvotes
```

Comment:
It looks like you are finding the movie object a number of times in this controller. You could create a `before_action` method that would find the movie for you before each method. Example:

`before_action :find_movie`

`private`

`def find_movie`
`@movie = Movie.find(params[:id])`
`end`

And then you could reference the `@movie` in your subsequent methods.

Ex (instead of lines 46-49 above): 

`def upvote`
`current_score = @movie.upvotes`
...